proposals/text/0000-evaluation-haystack-2.md

```diff
@@ -0,0 +1,223 @@
+- Title: Evaluation in Haystack 2.0
+- Decision driver: (Silvano Cerza, Julian Risch)
+- Start Date: 2023-08-23
+- Proposal PR: #5794
+- Github Issue or Discussion: https://github.com/deepset-ai/haystack/issues/5265
+
+# Summary
+
+Given the below requirements we redefine how evaluation and metrics are handled in Haystack 2.x.
+
+Our goal is to lower the barrier of entry for new comers but also making it more flexible and extensible for more advanced and expert users.
+All this while making it more modular and easier to test and maintain.
+
+The requirements are:
+
+- compare the performance of different pipelines on level of pipeline outputs (user perspective, integrated eval)
+  - while running the full pipeline we can store intermediate results and calculate metrics for each component that returns answers or documents
+- find out which component is the performance bottleneck in one pipeline by evaluating subpipelines (isolated evaluation)
+- as above, get evaluation metrics for every component in a pipeline that returns or documents (ranker, retriever, reader, PromptNode)
+- compare the performance of two components, for example two Readers, without the need to create a full retriever-reader pipeline
+- export evaluation results to a file (similar to Haystack 1.x but faster) and evaluation report
+- choose evaluation metrics from a list of metrics (e.g. F1, BLEU, ROUGE, Semantic Answer Similarity) based on the output type of a component
+- evaluate pipelines that return ExtractiveAnswers
+- evaluate pipelines that return GenerativeAnswers
+- evaluate hallucinations (check generated answers are backed up by retrieved documents)
+- evaluate pipelines with PromptNodes and arbitrary PromptTemplates (for example with Semantic Answer Similarity or BLEU, ROUGE (metrics from machine translation and summarization) if I provide labels)
+- load evaluation data for example from BEIR
```

Comment:
Do we plan to make the load interface generic? This can help with implementing your own custom loaders, e.g., loading data from an annotation tool, or, from HF datasets.

Reply:
I think my response to Malte's comment answers this. 

https://github.com/deepset-ai/haystack/pull/5794#issuecomment-1717684293